### PR TITLE
Redirect old blogs domain images to the new blogs domain

### PR DESCRIPTION
**Pull request by JakeChampion**
The old blogs domain is still being used for images and the domain no longer exists. We should replace the old domain with the new domain to make these images still work.

Currently we get roughly 4 thousand image requests for the old domain each day.

### CODE REVIEW

File: lib/routes/v2/images.js [hunk 1]
```diff
@@ -27,6 +27,17 @@ module.exports = app => {
 		next();
 	}
 
+	function updateOldBlogsLinks(request, response, next) {
+		if (request.params.schemeUrl.startsWith('https://blogs.r.ftdata.co.uk')) {
```
Comment (JakeChampion) [inline lib/routes/v2/images.js:31]:
```suggestion
		if (request.params.schemeUrl.startsWith('https://blogs.r.ftdata.co.uk/')) {
```

File: lib/routes/v2/images.js [hunk 2]
```diff
@@ -27,6 +27,17 @@ module.exports = app => {
...
+			request.params.schemeUrl = request.params.schemeUrl.replace('https://blogs.r.ftdata.co.uk', 'https://blogs.ft.com');
+			request.params.imageUrl = request.params.schemeUrl;
+		} else if (request.params.schemeUrl.startsWith('http://blogs.r.ftdata.co.uk')) {
```
Comment (JakeChampion) [inline lib/routes/v2/images.js:34]:
```suggestion
		} else if (request.params.schemeUrl.startsWith('http://blogs.r.ftdata.co.uk/')) {
```

**Comment by notlee** [comment]
Do you know where the requests are coming from? I'm wondering if there's one main source of outdated requests we could update? I'm not convinced mapping old domains should be the image services responsibility.

**Comment by JakeChampion** [comment]
> Do you know where the requests are coming from? I'm wondering if there's one main source of outdated requests we could update? I'm not convinced mapping old domains should be the image services responsibility.

They come from a bunch of old blog articles which are not going to be worked on. I already checked to see if we could get them updated at the source and it looks like no because they are decommissioning that system soon.

File: test/integration/v2/images-raw.test.js [hunk 3]
```diff
@@ -31,11 +31,25 @@ const testImageUris = {
...
+	oldLiveBlogsDomainHttp: 'http://blogs.ft.com/tech-blog/files/2012/03/Screen-Shot-2012-03-01-at-11.25.02-PM-391x270.png',
+	oldLiveBlogsDomainHttps: 'https://blogs.ft.com/tech-blog/files/2012/03/Screen-Shot-2012-03-01-at-11.25.02-PM-391x270.png',
 };
 
 describe('GET /v2/images/raw…', function() {
 
+	describe('/http://blogs.r.ftdata.co.uk', function() {
+		setupRequest('GET', `/v2/images/raw/${testImageUris.oldLiveBlogsDomainHttp}?source=test`);
+		itRespondsWithStatus(200);
+		itRespondsWithContentType('image/png');
```
Comment (JakeChampion) [inline test/integration/v2/images-raw.test.js:44]:
```suggestion
		itRespondsWithContentType('image/jpeg');
```

File: test/integration/v2/images-raw.test.js [hunk 4]
```diff
@@ -31,11 +31,25 @@ const testImageUris = {
...
+	});
+
+	describe('/https://blogs.r.ftdata.co.uk', function() {
+		setupRequest('GET', `/v2/images/raw/${testImageUris.oldLiveBlogsDomainHttps}?source=test`);
+		itRespondsWithStatus(200);
+		itRespondsWithContentType('image/png');
```
Comment (JakeChampion) [inline test/integration/v2/images-raw.test.js:50]:
```suggestion
		itRespondsWithContentType('image/jpeg');
```

**Comment by notlee** [comment]
>They come from a bunch of old blog articles which are not going to be worked on. I already checked to see if we could get them updated at the source and it looks like no because they are decommissioning that system soon.

Thanks for the context. I'm guessing these have started 404ing since we began removing images from Cloudinary's storage? Could we document what system this is for and maybe create an issue to remove the redirect when the system is decommissioned? Although I guess there's the chance new systems will come to depend on the redirect.

**Comment by JakeChampion** [comment]
> Thanks for the context. I'm guessing these have started 404ing since we began removing images from Cloudinary's storage? Could we document what system this is for and maybe create an issue to remove the redirect when the system is decommissioned? Although I guess there's the chance new systems will come to depend on the redirect.

What do you think of this? https://github.com/Financial-Times/origami-image-service/pull/641/files#diff-12383274d6f316ab2f3f636db95a140bR30-R32

**Comment by JakeChampion** [comment]
> Thanks for the context. I'm guessing these have started 404ing since we began removing images from Cloudinary's storage? Could we document what system this is for and maybe create an issue to remove the redirect when the system is decommissioned? Although I guess there's the chance new systems will come to depend on the redirect.

Yep, if this is accepted and merged we will create an issue to figure out how to decommission it. I have a splunk search which reports how many of these requests come through, when that number drops dramatically we can have it alert us to decommission this hack.

**Review by notlee — APPROVED** [review]
sounds great 👍

**Pull request merged**